src/components/Chat/BottomSheetGif.js

```diff
@@ -48,15 +41,34 @@ const BottomSheetGif = React.forwardRef((props, ref) => {
     setIsLoading(false);
   };
 
+  const searchGIF = React.useCallback(
+    debounce((search) => {
+      onLoad(search);
+    }, 500),
+    []
+  );
+
+  const handleChangeText = (text) => {
+    setSearchText(text);
+    if (text?.trim()?.length > 2) {
+      searchGIF(text);
+    }
+  };
+
+  const handleOnClearText = () => {
+    setSearchText('');
+    searchGIF('');
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The introduction of the `debounce` function in `searchGIF` is a good practice to avoid unnecessary API calls during user input. The `handleChangeText` and `handleOnClearText` functions have been updated to use this debounced function, which is a good approach. However, there's a potential issue with the condition `text?.trim()?.length > 2`. If `text` is null or undefined, calling `trim()` on it will throw an error. You should first check if `text` is truthy before calling `trim()`.
```diff
- const handleChangeText = (text) => {
+ const handleChangeText = (text) => {
    setSearchText(text);
-   if (text?.trim()?.length > 2) {
+   if (text && text.trim().length > 2) {
      searchGIF(text);
    }
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->